hw/ip/otbn/doc/fcov.md

```diff
@@ -0,0 +1,238 @@
+---
+title: OTBN functional coverage plan
+---
+
+This document describes our plan for functional coverage in OTBN.
+Some of this will probably be subsumed by a project-wide document in the future.
+As well as notes about how we're coming up up with functional coverage points for the processor, this document also outlines what those points will be.
+Eventually, much of this should be auto-generated, but we're starting with something manual.
+
+We distinguish between *architectural* and *micro-architectural* functional coverage.
+The idea is that the points that go into architectural coverage are those that a DV engineer could derive by reading the block specification.
+The points that go into micro-architectural coverage are those that require knowledge of the block's micro-architecture.
+Some of these will come from DV engineers; others from the block's designers.
+These two views are complementary and will probably duplicate coverage points.
+For example, an architectural coverage point might be "the processor executed `ADDI` and the result overflowed".
+This might overlap with something like "the `overflow` signal in the ALU was true when adding".
+
+# Block-based coverage
+
+## Call stack
+
+The [call stack]({{< relref ".#call-stack" >}}) is exposed as a special register behind `x1`.
+It has a bounded depth of 8 elements.
+We expect to see the following events:
+
+- Push to the call stack
+- Pop from the call stack
+- Push and pop from the call stack on a single instruction
```

Comment:
Push/Pop to/from full/empty call stack. Including the simultaneous push and pop when full (which should be allowed).

Reply:
I think this is covered by the next sentence?

Reply:
Ah yes, somehow missed that sentence.

Have thought of another one, double call stack access (e.g. add x2, x1, x1). Will only produce a single pop so both input operands get the same value.

Reply:
Ooh, good point! Added.